src/raster/processing/dem/options.rs

```diff
@@ -0,0 +1,172 @@
+    }
+
+    pub fn as_ptr(&self) -> *const GDALDEMProcessingOptions {
+        self.0.as_ptr()
+    }
+}
+
+impl Drop for GdalDEMProcessingOptions<'_> {
+    fn drop(&mut self) {
+        unsafe { GDALDEMProcessingOptionsFree(self.0.as_ptr()) };
+    }
+}
+
+/// DEM processor mode, to stringify and pass to [`gdal_sys::GDALDEMProcessing`].
+#[derive(Debug, Clone, Copy)]
+pub enum DemAlg {
+    Aspect,
+    ColorRelief,
+    Hillshade,
+    Roughness,
+    Slope,
+    Tpi,
+    Tri,
+}
+
+impl Display for DemAlg {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::ColorRelief => f.write_str("color-relief"),
+            _ => {
+                let s = format!("{self:?}").to_lowercase();
+                f.write_str(&s)
+            }
+        }
+    }
+}
+
+/// Slope and slope-related (aspect, hillshade) processing algorithms.
+///
+/// The literature suggests `ZevenbergenThorne` to be more suited to smooth landscapes,
+/// whereas `Horn` performs better on rougher terrain.
+#[derive(Debug, Clone, Copy)]
+pub enum DemSlopeAlg {
+    Horn,
+    ZevenbergenThorne,
+}
+
+impl Display for DemSlopeAlg {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.write_fmt(format_args!("{self:?}"))
+    }
+}
+
+macro_rules! common_dem_options {
+    () => {
+        /// Specify which band in the input [`Dataset`][crate::Dataset] to read from.
+        ///
+        /// Defaults to the first band.
+        pub fn with_input_band(&mut self, band: NonZeroUsize) -> &mut Self {
+            self.input_band = Some(band);
+            self
+        }
+
+        /// Fetch the specified input band to read from.
+        pub fn input_band(&self) -> Option<NonZeroUsize> {
+            self.input_band
+        }
+
+        /// Explicitly specify output raster format.
+        ///
+        /// This is equivalent to the `-of <format>` CLI flag accepted by many GDAL tools.
+        ///
+        /// The value of `format` must be the identifier of a driver supported by the runtime
+        /// environment's GDAL library (e.g. `COG`, `JPEG`, `VRT`, etc.). A list of these identifiers
+        /// is available from `gdalinfo --formats`:
+        ///
+        /// ```text
+        /// ❯ gdalinfo --formats
+        /// Supported Formats:
+        ///   VRT -raster,multidimensional raster- (rw+v): Virtual Raster
+        ///   DERIVED -raster- (ro): Derived datasets using VRT pixel functions
+        ///   GTiff -raster- (rw+vs): GeoTIFF
+        ///   COG -raster- (wv): Cloud optimized GeoTIFF generator
+        ///   NITF -raster- (rw+vs): National Imagery Transmission Format
+        /// ...
+        /// ```
+        ///
+        pub fn with_output_format(&mut self, format: &str) -> &mut Self {
+            self.output_format = Some(format.to_owned());
+            self
+        }
+
+        /// Fetch the specified output format driver identifier.
+        pub fn output_format(&self) -> Option<String> {
```

Comment:
I think it's fine to return `&str` for these, the caller probably won't care that much.

Reply:
https://github.com/metasim/gdal/pull/1/commits/7d2e1dd6a8a0fc95e3497c521379daffedcd1ec4